File: react-native-nitro-cache-module/ios/HybridNitroCacheModule.swift
import Foundation
import NitroModules

struct CacheEntry: Codable {
  let value: String
  let expireAt: Double?
}

class HybridNitroCacheModule: HybridNitroCacheModuleSpec {

  static let shared = HybridNitroCacheModule()
  private var cache: [String: CacheEntry] = [:]

  private var storageURL: URL {
    let docs = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
    return docs.appendingPathComponent("nitro_cache.json")
  }

  override init() {
    super.init()
    loadAllFromDisk()
  }

  private func loadAllFromDisk() {
    guard let data = try? Data(contentsOf: storageURL) else { return }
    guard let dict = try? JSONDecoder().decode([String: CacheEntry].self, from: data) else { return }
    let now = Date().timeIntervalSince1970 * 1000
    for (key, entry) in dict {
      if let expireAt = entry.expireAt, now > expireAt { continue }
      cache[key] = entry
    }
    print("Diskten yüklendi: \(cache.count) kayıt")
  }

  private func saveToDisk() {
    guard let data = try? JSONEncoder().encode(cache) else { return }
    try? data.write(to: storageURL, options: .atomic)
  }

  func setItem(key: String, value: String, ttl: Double) throws -> Void {
    let expireAt: Double? = ttl > 0 ? Date().timeIntervalSince1970 * 1000 + ttl : nil
    cache[key] = CacheEntry(value: value, expireAt: expireAt)
    saveToDisk()
    print("Kaydedildi: \(key)")
  }

  func getItem(key: String) throws -> Variant_NullType_String {
    guard let entry = cache[key] else {
      return .first(NullType.null)
    }
    if let expireAt = entry.expireAt {
      let now = Date().timeIntervalSince1970 * 1000
      if now > expireAt {
        cache.removeValue(forKey: key)
        saveToDisk()
        return .first(NullType.null)
      }
    }
    return .second(entry.value)
  }

  func removeItem(key: String) throws -> Void {
    cache.removeValue(forKey: key)
    saveToDisk()
    print("Silindi: \(key)")
  }

  func clear() throws -> Void {
    cache.removeAll()
    saveToDisk()
    print("Temizlendi!")
  }
}

File: react-native-nitro-cache-module/android/src/main/java/com/nitrocachemodule/HybridNitroCacheModule.kt
cat > /Users/ademalkan/Desktop/TodoApp/node_modules/react-native-nitro-cache-module/android/src/main/java/com/nitrocachemodule/HybridNitroCacheModule.kt << 'EOF'
package com.nitrocachemodule

import android.content.Context
import com.margelo.nitro.NitroModules
import com.margelo.nitro.core.NullType
import com.margelo.nitro.nitrocachemodule.HybridNitroCacheModuleSpec
import com.margelo.nitro.nitrocachemodule.Variant_NullType_String
import org.json.JSONObject
import java.io.File

class HybridNitroCacheModule : HybridNitroCacheModuleSpec() {

  private val cache = mutableMapOf<String, Pair<String, Double?>>()

  private val storageFile: File by lazy {
    File(NitroModules.applicationContext!!.filesDir, "nitro_cache.json")
  }

  init {
    loadAllFromDisk()
  }

  private fun loadAllFromDisk() {
    try {
      if (!storageFile.exists()) return
      val json = JSONObject(storageFile.readText())
      val now = System.currentTimeMillis().toDouble()
      for (key in json.keys()) {
        val entry = json.getJSONObject(key)
        val value = entry.getString("value")
        val expireAt = if (entry.has("expireAt")) entry.getDouble("expireAt") else null
        if (expireAt != null && now > expireAt) continue
        cache[key] = Pair(value, expireAt)
      }
      println("Diskten yüklendi: ${cache.size} kayıt")
    } catch (e: Exception) {
      println("loadAllFromDisk hatası: $e")
    }
  }

  private fun saveToDisk() {
    try {
      val json = JSONObject()
      for ((key, pair) in cache) {
        val entry = JSONObject()
        entry.put("value", pair.first)
        if (pair.second != null) entry.put("expireAt", pair.second)
        json.put(key, entry)
      }
      storageFile.writeText(json.toString())
    } catch (e: Exception) {
      println("saveToDisk hatası: $e")
    }
  }

  override fun setItem(key: String, value: String, ttl: Double?) {
    val expireAt = if (ttl != null && ttl > 0) System.currentTimeMillis() + ttl else null
    cache[key] = Pair(value, expireAt)
    saveToDisk()
    println("Kaydedildi: $key")
  }

  override fun getItem(key: String): Variant_NullType_String {
    val entry = cache[key] ?: return Variant_NullType_String.First(NullType.NULL)
    val expireAt = entry.second
    if (expireAt != null && System.currentTimeMillis() > expireAt) {
      cache.remove(key)
      saveToDisk()
      return Variant_NullType_String.First(NullType.NULL)
    }
    return Variant_NullType_String.Second(entry.first)
  }

  override fun removeItem(key: String) {
    cache.remove(key)
    saveToDisk()
    println("Silindi: $key")
  }

  override fun clear() {
    cache.clear()
    saveToDisk()
    println("Temizlendi!")
  }
}
EOF
